src/back/space/src/api/introspect.cpp

```diff
@@ -0,0 +1,95 @@
+#include "introspect.hpp"
+#include "../service/service.hpp"
+#include "../model/model.hpp"
+#include <shared/headers.hpp>
+#include <shared/errors.hpp>
+
+#include <boost/uuid/uuid_io.hpp>
+
+#include <userver/utils/boost_uuid4.hpp>
+#include <userver/http/common_headers.hpp>
+
+namespace svetit::space::handlers {
+
+Introspect::Introspect(
+	const components::ComponentConfig& conf,
+	const components::ComponentContext& ctx)
+	: server::handlers::HttpHandlerBase{conf, ctx}
+	, _s{ctx.FindComponent<Service>()}
+{}
+
+std::string Introspect::HandleRequestThrow(
+	const server::http::HttpRequest& req,
+	server::request::RequestContext&) const
+{
+	try {
+		// проверяем и получаем заголовок с ключом Пространства
+		if (!req.HasHeader("X-Original-URI"))
+			throw errors::BadRequest400("Header X-Original-URI is missing");
+		const std::string header = req.GetHeader("X-Original-URI");
+
+		// извлекаем ключ Пространства из заголовка
+		const std::string spaceKey = _s.GetKeyFromHeader(header);
+
+		// генерируем имя куки и проверяем наличие
+		const std::string cookieName = _s.GenerateCookieName(spaceKey);
+
+		if (req.HasCookie(cookieName)) {
+			// извлекаем токен из куки
+			const std::string token = req.GetCookie(cookieName);
+			LOG_DEBUG() << "Cookie exists, token: " << token;
+
+			try {
+				// проверяем и распарсиваем токен
+				SpaceTokenPayload data = _s.Tokens().Verify(token);
+				LOG_DEBUG() << "Token verify OK";
+
+				// авторизуем и отдаём заголовки
+				req.GetHttpResponse().SetHeader(headers::kSpaceId, data._id);
+				req.GetHttpResponse().SetHeader(headers::kSpaceRole, data._role);
+				return "Ok";
```

Comment:
А если вместо Ок пустую строку передавать, нормально отработает?
```suggestion
				return {};
```

Reply:
Готово - работает корректно.